src/util/options.rs

```diff
@@ -254,6 +255,96 @@ macro_rules! options {
     ]
 }
 
+#[derive(Clone, Debug, PartialEq)]
+/// AffinityKind describes how to set the affinity of GC threads. Note that we currently assume
+/// that each GC thread is equivalent to an OS or hardware thread.
+pub enum AffinityKind {
+    /// Delegate thread affinity to the OS scheduler
+    OsDefault,
+    /// Assign thread affinities over a list of cores in a round robin fashion. Note that if number
+    /// of threads > number of cores specified, then multiple threads will be assigned the same
+    /// core.
+    // XXX: Maybe using a u128 bitvector with each bit representing a core is more performant?
+    RoundRobin(Vec<CoreId>),
+}
+
+impl AffinityKind {
+    /// Returns an AffinityKind or String containing error. Expects the list of cores to be
+    /// formatted as numbers separated by commas, including ranges. There should be no spaces
+    /// between the cores in the list. For example: 0,5,8-11 specifies that the cores 0,5,8,9,10,11
+    /// should be used for pinning threads. Performs de-duplication of specified cores. Note that
+    /// the core list is sorted as a side-effect whenever a new core is added to the set.
+    fn parse_cpulist(cpulist: &str) -> Result<AffinityKind, String> {
```

Comment:
Should add unit tests for parsing.

Reply:
Yes I will make some unit tests

Reply:
I have made tests but there is a worry that the tests may spuriously fail in the CI as some of the tests depend on the existence of more than one core. If that's not true then the test may fail since the `validate()` function will fail.

Reply:
Actually I can directly test the parsing function instead of setting the environment variable, so this is not an issue.

Reply:
> I have made tests but there is a worry that the tests may spuriously fail in the CI as some of the tests depend on the existence of more than one core. If that's not true then the test may fail since the `validate()` function will fail.

Does that mean the `validate()` function will fail if there is only one core? That sounds like a bug in the code, rather than about the tests.

Reply:
No let me clarify. If I do `MMTK_THREAD_AFFINITY="0-1"` but I only have 1 core available (either it's a single core machine or I'm running `taskset -c 0` on the program), then the `validate()` function will rightfully say hey I don't know what core id is `1` so this is not a valid core list.

Reply:
Also you could get the number of CPUs, and use that to generate a valid test input.

Reply:
Yes I was making this exact test. But I've also changed most of the tests to directly test the parsing function which is equivalent imo.